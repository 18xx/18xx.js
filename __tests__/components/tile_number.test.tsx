import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Hexagon from '../../src/hexagon';
import Point from '../../src/point';

import Tile from '../../src/components/tile';
import TileNumber, { TileNumberProps } from '../../src/components/tile_number';

describe('TileNumber', () => {
  const hexagon: Hexagon = new Hexagon('east-west');

  describe('#render()', () => {
    const props: TileNumberProps = {
      hexagon,
      num: '54',
    };

    describe('when providing a rotation number', () => {
      it('returns the svg for the tile', () => {
        const subject: ReactElement<TileNumber> = (
          <TileNumber {...props} orientation={2} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when not providing a rotation number', () => {
      it('returns the svg for the tile', () => {
        const subject: ReactElement<TileNumber> = (
          <TileNumber {...props} />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when at a specific point', () => {
      const point: Point = new Point(5, 6);

      describe('when the hexagon is east-west orientation', () => {
        it('returns the svg for the tile', () => {
          const subject: ReactElement<TileNumber> = (
            <TileNumber {...props} point={point} />
          );
          expect(renderer.create(subject)).toMatchSnapshot();
        });
      });

      describe('when the hexagon is north-south orientation', () => {
        it('returns the svg for the tile', () => {
          const nsHex: Hexagon = new Hexagon('north-south');
          const subject: ReactElement<TileNumber> = (
            <TileNumber {...props} hexagon={nsHex} />
          );
          expect(renderer.create(subject)).toMatchSnapshot();
        });
      });

      describe('when the point is at the tile center', () => {
        it('returns the svg for the tile', () => {
          const subject: ReactElement<TileNumber> = (
            <TileNumber {...props} point={Tile.CENTER} />
          );
          expect(renderer.create(subject)).toMatchSnapshot();
        });
      });
    });
  });
});
