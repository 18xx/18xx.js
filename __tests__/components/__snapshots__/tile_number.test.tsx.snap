// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileNumber #render() when at a specific point when the hexagon is east-west orientation returns the svg for the tile 1`] = `
<text
  fill="black"
  fontSize={8}
  textAnchor="start"
  x={5}
  y={6}
>
  54
</text>
`;

exports[`TileNumber #render() when at a specific point when the hexagon is north-south orientation returns the svg for the tile 1`] = `
<text
  fill="black"
  fontSize={8}
  textAnchor="end"
  x={96}
  y={108.85125168440814}
>
  54
</text>
`;

exports[`TileNumber #render() when at a specific point when the point is at the tile center returns the svg for the tile 1`] = `
<text
  fill="black"
  fontSize={8}
  textAnchor="middle"
  x={55.42562584220407}
  y={64}
>
  54
</text>
`;

exports[`TileNumber #render() when not providing a rotation number returns the svg for the tile 1`] = `
<text
  fill="black"
  fontSize={8}
  textAnchor="end"
  x={109.85125168440814}
  y={94}
>
  54
</text>
`;

exports[`TileNumber #render() when providing a rotation number returns the svg for the tile 1`] = `
<text
  fill="black"
  fontSize={8}
  textAnchor="end"
  x={109.85125168440814}
  y={94}
>
  54.2
</text>
`;
